include/string_span.h

```diff
@@ -22,6 +22,16 @@
 #include "span.h"
 #include <cstring>
 
+// VS 2013 workarounds
+#ifdef _MSC_VER
+#if _MSC_VER <= 1800
+
+#pragma push_macro("GSL_MSVC_HAS_TYPE_DEDUCTION_BUG") 
```

Comment:
Do you expect many definitions of `GSL_MSVC_HAS_TYPE_DEDUCTION_BUG`?
If not, then the push here will hide conflicts that should be loudly reported to consumers if they ever step on it.